creators_custom/RoutesCreatorFenix.py

```diff
@@ -0,0 +1,650 @@
+        elif len(result.get_relations()) == 0:
+            sys.stderr.write("No route %s found" % route_ref)
+            return
+
+        for rel in result.get_relations():
+            print self.get_route(routes, route_masters, rel, warn=False)
+
+        self.save_routes_to_file(routes)
+
+    def get_stops_of_route(self, route_id):
+        api = overpy.Overpass()
+
+        result = api.query("""
+        <id-query ref="%s" type="relation"/>
+        <print/>
+        """ % str(route_id))
+
+        if len(result.get_relations()) == 0:
+            raise RuntimeError(
+                "Route not found: https://www.openstreetmap.org/relation/" + str(route_id))
+
+        stops = []
+
+        for member in result.get_relation(route_id, resolve_missing=True).members:
+            if member.role.startswith('platform'):
+                stop = self.from_relation_member(member)
+                stops.append(stop)
+            elif CHECK_FOR_NON_PLATFORMS:
+                # It is possible bus_stops are not properly set to role=platform*
+
+                if member.role == "forward" or member.role == "backward":
+                    continue
+
+                print "Looking for bus stops without proper role in relation..."
+
+                complete_member = member.resolve(resolve_missing=True)
+                tags = complete_member.tags
+
+                if isinstance(member, overpy.RelationNode):
+                    if 'highway' in tags and tags['highway'] == 'bus_stop':
+                        stop = self.from_relation_member(member)
+                        stops.append(stop)
+                        sys.stderr.write("Node has no platform* role in relation " +
+                                         str(route_id) + ": https://www.openstreetmap.org/node/" +
+                                         str(member.ref) + "\n")
+                elif isinstance(member, overpy.RelationWay):
+                    if 'amenity' in tags and tags['amenity'] == 'bus_station':
+                        stop = self.from_relation_member(member)
+                        stops.append(stop)
+                        sys.stderr.write("Way has no platform* role in relation " + str(route_id) +
+                                         ": https://www.openstreetmap.org/way/" + str(member.ref) +
+                                         "\n")
+
+        return stops
+
+    def from_relation_member(self, member):
+        stop_id = member.ref
+
+        if isinstance(member, overpy.RelationNode):
+            stop_type = "node"
+        elif isinstance(member, overpy.RelationWay):
+            stop_type = "way"
+        else:
+            raise RuntimeError("Unknown type: " + str(member))
+
+        return Stop(stop_id, stop_type)
+
+    def save_routes_to_file(self, outes):
+        routes = creators_custom.ROUTES
+        with open('data/routes.pkl', 'wb') as f:
+            pickle.dump(routes, f, pickle.HIGHEST_PROTOCOL)
```

Comment:
This was my proposed workaround until we discussed #26. Let me try to describe the problem (I hope it's understandable): The current code creates a parallel data structure by using custom classes (RouteMaster, Route, Stop) and those are logically connected, which means that they are potentially created in the RoutesCreator, but used in StopsCreator and TripsCreator. I don't think we can resolve this in this PR, especially if we want to go step by step. Therefore I thought putting it into a global variable (just for now) would be a solution until we go into this part.